node/src/actors/chain_manager/handlers.rs

```diff
@@ -1409,6 +1408,84 @@ impl Handler<GetBalance> for ChainManager {
     }
 }
 
+impl Handler<GetSupplyInfo> for ChainManager {
+    type Result = Result<SupplyInfo, failure::Error>;
+
+    fn handle(&mut self, _msg: GetSupplyInfo, _ctx: &mut Self::Context) -> Self::Result {
+        if self.sm_state != StateMachine::Synced {
+            return Err(ChainManagerError::NotSynced {
+                current_state: self.sm_state,
+            }
+            .into());
+        }
+
+        let total_supply = 2_500_000_000_000_000_000;
```

Comment:
The total supply can be calculated using the other fields:

```rust
let genesis_amount = current_locked_supply + current_unlocked_supply + locked_wits_by_requests - blocks_minted_reward;
let total_block_reward = total_block_reward(
    consensus_constants.initial_block_reward,
    consensus_constants.halving_period,
);
let total_supply = genesis_amount + total_block_reward;
let total_supply_minus_missing_blocks = total_supply - blocks_missing_reward;
```

Or it can be hardcoded, although a more accurate value would be

```
let total_supply = 750000000007667720 + 1749999999954500000;
```